build_projects/dotnet-cli-build/GenerateChecksums.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+using System.Security.Cryptography;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GenerateChecksums : Task
+    {
+        /// <summary>
+        /// An item collection of files for which to generate checksums.  Each item must have metadata
+        /// 'DestinationPath' that specifies the path of the checksum file to create.
+        /// </summary>
+        [Required]
+        public ITaskItem[] Items { get; set; }
```

Comment:
I thought about implementing it like that, but decided to follow the pattern used by the `UploadToAzure` and `SetBlobPropertiesBasedOnFileType` tasks, which each rely on `RelativeBlobPath` metadata being set.  I think the documentation on the Items variable plus the error message if it's not set make it discoverable enough for clients.